pkg/controller/core/workload_controller.go

```diff
@@ -893,6 +889,14 @@ func (r *WorkloadReconciler) admittedNotReadyWorkload(wl *kueue.Workload) (strin
 	return "", 0
 }
 
+func (r *WorkloadReconciler) evictWorkload(ctx context.Context, wl *kueue.Workload, baseReason, reason, message, underlyingCause string) (bool, error) {
```

Comment:
Can we move this function to the `pkg/workload/workload.go` package and pass client as one of the parameters?